store/store.go

```diff
@@ -227,25 +229,23 @@ func (s *DefaultStore) LoadValidators(height uint64) (*tmtypes.ValidatorSet, err
 }
 
 // loadHashFromIndex returns the hash of a block given its height
-func (s *DefaultStore) loadHashFromIndex(height uint64) ([32]byte, error) {
+func (s *DefaultStore) loadHashFromIndex(height uint64) (header.Hash, error) {
 	blob, err := s.db.Get(s.ctx, ds.NewKey(getIndexKey(height)))
 
-	var hash [32]byte
 	if err != nil {
-		return hash, fmt.Errorf("failed to load block hash for height %v: %w", height, err)
+		return nil, fmt.Errorf("failed to load block hash for height %v: %w", height, err)
 	}
-	if len(blob) != len(hash) {
-		return hash, errors.New("invalid hash length")
+	if len(blob) != 32 {
```

Comment:
[nit] Instead of 32 can we specify this to the length of the type of header.Hash?

Reply:
No, as header.Hash is `[]byte`, but definitely this should be un-hardcoded later (we planed to support "pluggable" hashes anyways).